src/Native/Unix/System.Native/pal_io.cpp

```diff
@@ -577,7 +577,15 @@ extern "C" int32_t SystemNative_Link(const char* source, const char* linkTarget)
 extern "C" intptr_t SystemNative_MksTemps(char* pathTemplate, int32_t suffixLength)
 {
     intptr_t result;
+#if HAVE_MKSTEMPS
     while (CheckInterrupted(result = mkstemps(pathTemplate, suffixLength)));
+#elif
+    // mkstemps is not available bionic/Android, but mkstemp is
+    (void)suffixLength; // To prevent the compiler from generating errors because suffixLength is unused
+    while (CheckInterrupted(result = mkstemp(pathTemplate)));
```

Comment:
Yup, didn't catch the fact that it's about a suffix length.

So I guess the better option would be to create a new string without the suffix, pass that to mkstemp, and add back the suffix?

Alternatively we may also have the managed code differentiate between mkstemp and mkstemps , but that seems to be adding more overhead.